ibm/service/power/resource_ibm_pi_host.go

```diff
@@ -0,0 +1,372 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return resourceIBMPIHostRead(ctx, d, meta)
+}
+
+func resourceIBMPIHostRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, hostID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	client := instance.NewIBMPIHostGroupsClient(ctx, sess, cloudInstanceID)
+	host, err := client.GetHost(hostID)
+	if err != nil {
+		if strings.Contains(err.Error(), NotFound) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	d.Set(Attr_HostID, host.ID)
+
+	capacity := []map[string]interface{}{}
+	if host.Capacity != nil {
+		cp := PIHostCapacityToMap(host.Capacity)
+		capacity = append(capacity, cp)
+		d.Set(Attr_Capacity, capacity)
+	}
+	if host.DisplayName != "" {
+		d.Set(Attr_DisplayName, host.DisplayName)
+	}
+	if host.HostGroup != nil {
+		hostgroup := PIHostGroupToMap(host.HostGroup)
+		d.Set(Attr_HostGroup, hostgroup)
+	}
+	if host.State != "" {
+		d.Set(Attr_State, host.State)
+	}
+	if host.Status != "" {
+		d.Set(Attr_Status, host.Status)
+	}
+	if host.SysType != "" {
+		d.Set(Attr_SysType, host.SysType)
+	}
+
+	return nil
+}
+
+func resourceIBMPIHostUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, hostID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	displayName := d.Get(Arg_Host + ".0").(map[string]interface{})[Attr_DisplayName].(string)
+	client := instance.NewIBMPIHostGroupsClient(ctx, sess, cloudInstanceID)
+	if d.HasChange(Arg_Host) {
+
+		hostBody := models.HostPut{
+			DisplayName: &displayName,
+		}
+		_, err := client.UpdateHost(&hostBody, hostID)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	return resourceIBMPIHostRead(ctx, d, meta)
+}
+
+func resourceIBMPIHostDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, hostID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	client := instance.NewIBMPIHostGroupsClient(ctx, sess, cloudInstanceID)
+	err = client.DeleteHost(hostID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	_, err = isWaitForPIHostDeleted(ctx, client, hostID, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId("")
+
+	return nil
+}
+func isWaitForPIHostDeleted(ctx context.Context, client *instance.IBMPIHostGroupsClient, id string, timeout time.Duration) (interface{}, error) {
+	log.Printf("Waiting for host (%s) to be deleted.", id)
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{helpers.PIInstanceDeleting},
```

Comment:
```suggestion
		Pending:    []string{State_DELETING},
```
Replace with State variables?